tests/ccl_test_redshifts.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+from numpy.testing import assert_allclose, run_module_suite
+import numpy.testing
+import pyccl as ccl
+
+# Set tolerances
+TOLERANCE = 1e-4
+
+def check_redshift_analytic():
+    """
+    Compare the redshift functions to an analytic toy case.
+    """
+    # Redshift input
+    z_lst = [0., 0.5, 1., 1.5, 2.]
+
+    # p(z) function for which we can calculate and analytic dNdz_tomog
+    # when pairs with the toy dndz_ana below
+    def pz_ana(z_ph, z_s, args):
+        return (np.exp(- (z_ph - z_s)**2. / 2. / 0.1**2) / np.sqrt(2.
+                * np.pi) / 0.1)
+
+    # PhotoZFunction class
+    PZ_ana = ccl.PhotoZFunction(pz_ana)
+        
+    # Introduce an unrealistic, simple true function for dNdz for which
+    # we can calculate dNdz_tomog analytically for comparison.
+    def dndz_ana(z, args):
+        if ((z>=0.) and (z<=1.0)): 
+            return 1.0
+        else:
+            return 0. 
+	
+    # import math erf and erfc functions:
+    from math import erf, erfc
+    # And introduce a function to return the analytic result of using 
+    # dndz_ana with a Gaussian p(z,z'), to which we compare
+    def dNdz_tomog_analytic(z, sigz, zmin, zmax):
+        if ( (z>=0.) and (z<=1.0) ):
+            return (erf((z-zmin) / np.sqrt(2.)/sigz) - erf((z-zmax)/
+                    np.sqrt(2.)/sigz)) / (-1. + (-np.exp(-(zmax-1)**2 
+                    / 2. / sigz**2) + np.exp(-zmax**2 / 2. / sigz**2) + 
+                    np.exp(-(zmin-1)**2 / 2. / sigz**2) - np.exp(
+                    -zmin**2 / 2 /sigz**2)) * np.sqrt(2. / np.pi)*sigz 
+                    + erf( (zmax-1) / np.sqrt(2.) /  sigz) + erfc( 
+                    (zmin-1.)/np.sqrt(2.)/sigz) + zmax * (erf(zmax/
+                    np.sqrt(2.)/sigz) - erf( (zmax-1.) / np.sqrt(2.)/
+                    sigz)) + zmin * ( erf( (zmin-1.)/np.sqrt(2.)
+		            / sigz) - erf(zmin/np.sqrt(2.)/sigz)))
```

Comment:
Maybe mention where you got this.  I assume this is copied from a Wolfram Alpha output.  Maybe just give the link to this page where you had it do the integral.

Reply:
I actually did it in a mathematica notebook - I could include the notebook in the benchmarks folder?

Reply:
I don't think it's worth including the notebook.  Probably just put an inline comment here saying what the Mathematica integral was that you did of which this is the output.  I.e. enough that someone else can see what you did and reproduce if necessary.